src/cmd/flux-job.c

```diff
@@ -3427,23 +3427,19 @@ int cmd_info (optparse_t *p, int argc, char **argv)
         val = new_val = reconstruct_current_jobspec (jobspec, eventlog);
     }
     /* The current (non --base) R is obtained through the
-     * job-info.update-lookup RPC, not the normal job-info.lookup.
+     * job-info.lookup RPC w/ the CURRENT flag.
      */
     else if (!optparse_hasopt (p, "base") && streq (key, "R")) {
-        json_t *o;
         if (!(f = flux_rpc_pack (h,
-                                 "job-info.update-lookup",
+                                 "job-info.lookup",
                                  FLUX_NODEID_ANY,
                                  0,
-                                 "{s:I s:s s:i}",
+                                 "{s:I s:[s] s:i}",
                                  "id", id,
-                                 "key", key,
-                                 "flags", 0))
-            || flux_rpc_get_unpack (f, "{s:o}", key, &o) < 0)
+                                 "keys", key,
+                                 "flags", FLUX_JOB_LOOKUP_CURRENT))
```

Comment:
Use of this flag requires a current version of flux-core. I'm not sure if it is worth it, but should this call fall back to `update-lookup` if the RPC returns `EPROTO`?

Reply:
ugh, It ends up I didn't check "flags" in older versions b/c flags wasn't used.  So I guess a caller would just get the original version.

Could just keep on calling `job-info.update-lookup`?  But I'm not sure the pros of that outweigh the cons.  I figure most users, if they are using `flux job info`, base R == updated R anyways?

Reply:
>  I figure most users, if they are using flux job info, base R == updated R anyways?

Yes, there will be an updated R only if the expiration is adjusted, which is going to pretty rare. Also, the command would have to be run against a downrev parent instance to hit the failure, and at least this would be a failure I think instead of returning wrong information. So perhaps we just keep things as is and hope for the best? Over time it won't be a problem anymore... (However, something to keep in mind when adding future interfaces)